File: Leetcode/1209 Remove All Adjacent Duplicates in String II.ts
function removeDuplicates(s: string, k: number): string {
    const stack: [string, number][] = [];
    for(const c of s) {
        if(stack.length !== 0 && stack[stack.length - 1][0] === c) {
            stack[stack.length - 1][1]++;
        } else {
            stack.push([c, 1]);
        }
        if(stack[stack.length - 1][1] === k) {
            stack.pop();
        }
    }
    return stack.map(([c, n]) => c.repeat(n)).join("");
}

File: Leetcode/1209 Remove All Adjacent Duplicates in String II.py
class Solution:
    def removeDuplicates(self, s, k):
        stack = []
        for c in s:
            if stack and stack[-1][0] == c:
                stack[-1][1] += 1
            else:
                stack.append([c, 1])
            if stack[-1][1] == k:
                stack.pop()
        return "".join(map(lambda cn: cn[0]*cn[1], stack))
